include/node.h

```diff
@@ -0,0 +1,95 @@
+#ifndef PIPE_NETWORK_NODE_H_
+#define PIPE_NETWORK_NODE_H_
+
+#include <Eigen/Sparse>
+
+//! Node class
+//! \brief Class that stores the information about nodes
+class Node {
+
+	public:
+
+		// Constructor with id and coordinates
+		//! \param[in] id assign as the id_ of the node
+		//! \param[in] coords coordinates of the node
+		Node(unsigned id, const Eigen::Vector3d& coords) : id_{id} {
+			for (unsigned i = 0; i < coords.size(); i++) coordinates_[i] = coords(i);
+		}
+
+		//! Destructor
+		~Node() = default;
+
+		//! Copy constructor
+		Node(const Node&) = delete;
+
+		//! Assignment operator
+		Node& operator=(const Node&) = delete;
+
+		//! Move constructor
+		Node(Node&&) = delete;
+
+		//! Return id
+		//! \param[out] id_ return id of the node
```

Comment:
These should be `\retval` not `param[out]`, `param[out]` is used when it's passed in and out of the function as arguments. Here you are returning a variable.